src/transformers/models/hiera/modeling_hiera.py

```diff
@@ -0,0 +1,811 @@
+                unroll_schedule = unroll_schedule[1:]
+
+    def forward(self, x: torch.Tensor, block_idx: int, mask: torch.Tensor = None) -> torch.Tensor:
+        """
+        Roll the given tensor back up to spatial order assuming it's from the given block.
+
+        If no mask is provided:
+            - Returns [B, H, W, C] for 2d, [B, T, H, W, C] for 3d, etc.
+        If a mask is provided:
+            - Returns [B, #MUs, MUy, MUx, C] for 2d, etc.
+        """
+        schedule, size = self.schedule[block_idx]
+        B, N, C = x.shape
+
+        D = len(size)
+        cur_mu_shape = [1] * D
+
+        for strides in schedule:
+            # Extract the current patch from N
+            x = x.view(B, *strides, N // math.prod(strides), *cur_mu_shape, C)
+
+            # Move that patch into the current MU
+            # Example in 2d: [B, Sy, Sx, N//(Sy*Sx), MUy, MUx, C] -> [B, N//(Sy*Sx), Sy, MUy, Sx, MUx, C]
+            L = len(x.shape)
+            permute = (
+                [0, 1 + D]
+                + sum(
+                    [list(p) for p in zip(range(1, 1 + D), range(1 + D + 1, L - 1))],
+                    [],
+                )
+                + [L - 1]
+            )
+            x = x.permute(permute)
+
+            # Reshape to [B, N//(Sy*Sx), *MU, C]
+            for i in range(D):
+                cur_mu_shape[i] *= strides[i]
+            x = x.reshape(B, -1, *cur_mu_shape, C)
+            N = x.shape[1]
+
+        # Current shape (e.g., 2d: [B, #MUy*#MUx, MUy, MUx, C])
+        x = x.view(B, N, *cur_mu_shape, C)
+
+        # If masked, return [B, #MUs, MUy, MUx, C]
+        if mask is not None:
+            return x
+
+        # If not masked, we can return [B, H, W, C]
+        x = undo_windowing(x, size, cur_mu_shape)
+
+        return x
+
+
+@dataclass
+class HieraModelOutput(ModelOutput):
+    """
+    Base class for HieraModel model's outputs, conforming to Hugging Face's ModelOutput.
+
+    Args:
+        last_hidden_state (torch.FloatTensor of shape (batch_size, sequence_length, hidden_size)):
+            Last layer hidden-states.
+        intermediates (List[torch.Tensor], optional):
+            Intermediate representations or features from the model, if applicable.
+    """
+
+    last_hidden_state: torch.FloatTensor
+    intermediates: Optional[List[torch.Tensor]] = None
+
+
+class MaskUnitAttention(nn.Module):
+    """
+    Computes either Mask Unit or Global Attention. Also is able to perform q pooling.
+
+    Note: this assumes the tokens have already been flattened and unrolled into mask units.
+    See `Unroll` for more details.
+    """
+
+    def __init__(
+        self,
+        input_dim: int,
+        output_dim: int,
+        number_of_heads: int,
+        q_stride: int = 1,
+        window_size: int = 0,
+        use_mask_unit_attention: bool = False,
+    ):
+        """
+        Args:
+        - input_dim, output_dim: The input and output feature dimensions.
```

Comment:
No need to make a list with `-`

```suggestion
        input_dim (`int`): The input feature dimensions.
        output_dim (`int`): The output feature dimensions.
```